apis/bigquerydatatransfer/v1alpha1/bigquerydatatransferconfig_types.go

```diff
@@ -0,0 +1,197 @@
+	//  associated with this transfer config finish.
+	//
+	//  The format for specifying a pubsub topic is:
+	//  `projects/{project_id}/topics/{topic_id}`
+	NotificationPubsubTopic *string `json:"notificationPubsubTopic,omitempty"`
+
+	// Parameters specific to each data source. For more information see the
+	//  bq tab in the 'Setting up a data transfer' section for each data source.
+	//  For example the parameters for Cloud Storage transfers are listed here:
+	//  https://cloud.google.com/bigquery-transfer/docs/cloud-storage-transfer#bq
+	Params map[string]string `json:"params,omitempty"`
+
+	// +required
+	Parent `json:",inline"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable.
+	// The BigQueryDataTransferConfig name. If not given, the metadata.name will be used.
+	// + optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Data transfer schedule.
+	//  If the data source does not support a custom schedule, this should be
+	//  empty. If it is empty, the default value for the data source will be used.
+	//  The specified times are in UTC.
+	//  Examples of valid format:
+	//  `1st,3rd monday of month 15:30`,
+	//  `every wed,fri of jan,jun 13:15`, and
+	//  `first sunday of quarter 00:00`.
+	//  See more explanation about the format here:
+	//  https://cloud.google.com/appengine/docs/flexible/python/scheduling-jobs-with-cron-yaml#the_schedule_format
+	//
+	//  NOTE: The minimum interval time between recurring transfers depends on the
+	//  data source; refer to the documentation for your data source.
+	Schedule *string `json:"schedule,omitempty"`
+
+	// Options customizing the data transfer schedule.
+	ScheduleOptions *ScheduleOptions `json:"scheduleOptions,omitempty"`
+
+	// Deprecated. Unique ID of the user on whose behalf transfer is done.
+	UserID *int64 `json:"userID,omitempty"`
+}
+
+type Parent struct {
+	// +required
+	ProjectRef *refv1beta1.ProjectRef `json:"projectRef"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location field is immutable"
+	// Immutable.
+	// +required
+	Location string `json:"location"`
+}
+
+// BigQueryDataTransferConfigStatus defines the config connector machine state of BigQueryDataTransferConfig
+type BigQueryDataTransferConfigStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the BigQueryDataTransferConfig resource in GCP.
+	// +optional
```

Comment:
super nit: we can ignore the `+optional` since this is the default and the status is not a user specified field.

Reply:
These come from the controller builder template. Removing them in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2613.